blocks/main/latest-news/mascot/index.tsx

```diff
@@ -0,0 +1,111 @@
+import { memo, useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { useInView } from 'react-intersection-observer';
+import { AnimationItem, LottiePlayer } from 'lottie-web/build/player/lottie_svg';
+
+import styles from './mascot.module.css';
+
+type MascotProps = {
+    className?: string;
+};
+
+const ANIMATION_INITIAL_DELAY = 1500 as const;
+const ANIMATION_AFTER_DELAY = 7000 as const;
+
+function sleep(ms: number) {
+    return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+function createAnimation(
+    lottie: LottiePlayer,
+    node: Element,
+    animationData: Record<string, unknown>
+): [AnimationItem, () => Promise<void>] {
+    const animation = lottie.loadAnimation({
+        container: node,
+        renderer: 'svg',
+        loop: false,
+        autoplay: false,
+        animationData,
+    });
+
+    const animationComplete = new Promise((resolve) => {
+        function done() {
+            animation.removeEventListener('complete', done);
+            resolve(null);
+        }
+
+        animation.addEventListener('complete', done);
+    });
+
+    return [
+        animation,
+        async () => {
+            animation.play();
+            await animationComplete;
+            animation.destroy();
+        },
+    ];
+}
+
+function MascotContent({ className, onFinish }: MascotProps & { onFinish: () => void }) {
+    const mascotNode = useRef<HTMLSpanElement>(null);
+    const { ref: inViewRef, inView } = useInView();
+
+    useEffect(() => {
+        const node = mascotNode.current;
+
+        let animation: AnimationItem;
+        let play: ReturnType<typeof createAnimation>[1];
+
+        let isOnceStarted = false;
```

Comment:
nit: any chance that
```suggestion
        let wasOnceStarted = false;
```
sounds better?